src/EventListener/ReadListener.php

```diff
@@ -30,10 +32,11 @@
     private $collectionDataProvider;
     private $itemDataProvider;
 
-    public function __construct(CollectionDataProviderInterface $collectionDataProvider, ItemDataProviderInterface $itemDataProvider)
+    public function __construct(CollectionDataProviderInterface $collectionDataProvider, ItemDataProviderInterface $itemDataProvider, SubcollectionDataProviderInterface $subcollectionDataProvider)
```

Comment:
For all new arguments, maybe can you add deprecation notice if new deps are not passed so we'll be able to remove this code in API Platform 3?